flink-end-to-end-tests/flink-tpcds-test/src/main/java/org/apache/flink/table/tpcds/schema/TpcdsSchemaProvider.java

```diff
@@ -0,0 +1,526 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.tpcds.schema;
+
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.types.DataType;
+
+import java.sql.Date;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class to provide all TPC-DS tables' schema information.
+ * The data type of column use {@link DataType}
+ */
+public class TpcdsSchemaProvider {
+
+	private static int tpcdsTableNums = 24;
+	private static Map<String, TpcdsSchema> schemaMap = new HashMap<>(tpcdsTableNums);
+
+	static {
```

Comment:
Nit: I think it is a bit more clean to not use a static initializer block, but a static method:

```java
private static final Map<String, TpcdsSchema> schemaMap = createTabeSchemas();

private static Map<String, TpcdsSchema> createTabeSchemas() {
    final Map<String, TpcdsSchema> schemaMap = new HashMap<>(tpcdsTableNums);
    schemaMap.put("catalog_sales", new TpcdsSchema(
        Arrays.asList(
            new Column("cs_sold_date_sk", DataTypes.BIGINT()),
...
    )));
}
``´